File: src/main/resources/output/IFC4X3_python/Entities/IfcPropertyAbstraction/IfcProperty/IfcSimpleProperty/IfcPropertyEnumeratedValue/IfcPropertyEnumeratedValue.py
from typing import List

class IfcPropertyEnumeratedValue(IfcSimpleProperty):
    def __init__(self):
        self.EnumerationValues: List[IfcValue] = None
        self.EnumerationReference: IfcPropertyEnumeration = None

    # === WHERE CLAUSES ===
    # WR21 : NOT(EXISTS(EnumerationReference)) OR  NOT(EXISTS(EnumerationValues)) OR  (SIZEOF(QUERY(temp <* EnumerationValues | temp IN EnumerationReference.EnumerationValues)) = SIZEOF(EnumerationValues))

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcPropertyAbstraction/IfcProperty/IfcSimpleProperty/IfcPropertyEnumeratedValue/IfcPropertyEnumeratedValue.cs
public class IfcPropertyEnumeratedValue : IfcSimpleProperty
{
    public List<IfcValue> EnumerationValues { get; set; }
    public IfcPropertyEnumeration EnumerationReference { get; set; }

    // === WHERE CLAUSES ===
    // WR21 : NOT(EXISTS(EnumerationReference)) OR  NOT(EXISTS(EnumerationValues)) OR  (SIZEOF(QUERY(temp <* EnumerationValues | temp IN EnumerationReference.EnumerationValues)) = SIZEOF(EnumerationValues))
}
